### PR TITLE
Don't throw exception in list of entities with un-accessible items

### PR DESCRIPTION
**Pull request by amitaibu**
#239

If no IDs were requested, we should not throw an exception in case an
entity is un-accessible by the user.

### CODE REVIEW

**Comment by amitaibu** [comment]
Next is actually checking the code I just wrote, and writing tests to it :)

**Comment by amitaibu** [comment]
Update: Error seems related to entity-validator module. Checking

**Comment by e0ipso** [comment]
@amitaibu cool! Let me know.

File: plugins/restful/RestfulEntityBase.php [hunk 1]
```diff
@@ -128,8 +128,14 @@ public function getList() {
 
     $return = array();
 
+    // If no IDs were requested, we should not throw an exception in case an
+    // entity is un-accessible by the user.
+    $error_on_no_access = !empty($request['id']) || !empty($request['filter']['id']);
```
Comment (e0ipso) [inline plugins/restful/RestfulEntityBase.php]: We should be using the `isListRequest` helper function.

**Comment by e0ipso** [comment]
One minor comment and we can hit the green.

**Comment by e0ipso** [comment]
Is this still WIP? The title says so.

**Comment by amitaibu** [comment]
Yeah, still WIP -- I want to add tests.

**Comment by e0ipso** [comment]
That's a good idea :-)

**Comment by amitaibu** [comment]
@mateu-aguilo-bosch ready for review.

**Comment by amitaibu** [comment]
Actually I got the filter part wrong, we just need to check if no specific ID was requested. So no review yet :)

**Comment by amitaibu** [comment]
Ready for review.

**Comment by e0ipso** [comment]
Merged!

Thanks Amitai!

**Pull request closed without merging**